### ISSUE TITLE
junctions don't work with project.refs references

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#344]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/344)
In GitLab by [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) on Apr 6, 2018, 17:10

With `ref-storage: project.refs`, I get the following error, even though I used `bst track` and the ref is present in `project.refs`.

```
[00:00:02][][] FAILURE Fetching https://gitlab.com/freedesktop-sdk/freedesktop-sdk/
[00:00:02][][] FAILURE Loading pipeline
Error loading pipeline: git source at base/freedesktop-sdk-junction.bst [line 3 column 2]: expected ref 'None' was not found in git repository: 'https://gitlab.com/freedesktop-sdk/freedesktop-sdk/'
```

A workaround is to manually add the `ref` in the `.bst` file for the junction, but I really don't like that.

I've already mentioned it in https://gitlab.com/BuildStream/buildstream/issues/292#note_64566339 but I think this more important than the rest.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) on Apr 6, 2018, 17:17

changed the description

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 6, 2018, 17:46

Commented on wrong issue from handphone at 2am in bar it seems: please verify that this is fixed in master, i fixed this yesterday.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) on Apr 6, 2018, 19:09

Indeed, this was fixed in 857751fb6479afa0ee83d4759891f7c4837b87bb. Is a new release supposed to happen soon?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) on Apr 6, 2018, 19:09

closed
